lib/semian/simple_enum.rb

```diff
@@ -0,0 +1,40 @@
+require 'forwardable'
+
+module Semian
+  module Simple
+    class Enum #:nodoc:
+      extend Forwardable
+
+      def_delegators :@integer, :destroy
+
+      def initialize(symbol_list:)
+        @integer = ::Semian::Simple::Integer.new
+        initialize_lookup(symbol_list)
+      end
+
+      def increment(val = 1)
+        @integer.value = (@integer.value + val) % @sym_to_num.size
+        value
+      end
+
+      def value
+        @num_to_sym.fetch(@integer.value)
+      end
+
+      def value=(sym)
+        @integer.value = @sym_to_num.fetch(sym)
+      end
+
+      private
+
+      def initialize_lookup(symbol_list)
+        # Assume symbol_list[0] is mapped to 0
+        # Cannot just use #object_id since #object_id for symbols is different in every run
+        # For now, implement a C-style enum type backed by integers
+
+        @sym_to_num = Hash[symbol_list.each_with_index.to_a]
+        @num_to_sym = @sym_to_num.invert
+      end
+    end
+  end
+end
```

Comment:
No, the Sysv should throw an ArgumentError as well. `KeyError` for an Enum make 0 sense.

You are leaking implementation details to the caller.

Also it doesn't matter that much because this is a programming error exception, the kind you are not supposed to rescue.